Code/Framework/AzCore/AzCore/Time/ITime.h

```diff
@@ -13,14 +13,26 @@
 #include <AzCore/RTTI/RTTI.h>
 #include <AzCore/RTTI/TypeSafeIntegral.h>
 #include <AzCore/std/time.h>
+#include <AzCore/std/chrono/chrono.h>
 
 namespace AZ
 {
     //! This is a strong typedef for representing a millisecond value since application start.
     AZ_TYPE_SAFE_INTEGRAL(TimeMs, int64_t);
 
+    //! This is a strong typedef for representing a microsecond value since application start.
+    //! Using int64_t as the underlying type, this is good to represent approximately 292,471 years
+    AZ_TYPE_SAFE_INTEGRAL(TimeUs, int64_t);
```

Comment:
The changes related to this look clean! Are there any units that need to be updated/added to?

Reply:
Not at the moment, there really should be though. :(

Reply:
Added tests:
`[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from TimeTests
[ RUN      ] TimeTests.TestConversionUsToMs
[       OK ] TimeTests.TestConversionUsToMs (4 ms)
[ RUN      ] TimeTests.TestConversionMsToUs
[       OK ] TimeTests.TestConversionMsToUs (0 ms)
[ RUN      ] TimeTests.TestClocks
[       OK ] TimeTests.TestClocks (0 ms)
[----------] 3 tests from TimeTests (5 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test case ran. (9 ms total)
[  PASSED  ] 3 tests.`